test/libsolidity/semanticTests/libraries/library_delegatecall_guard_view_staticcall.sol

```diff
@@ -0,0 +1,31 @@
+contract D {
+    uint public x;
+    constructor() public { x = 42; }
+}
+library L {
+    function f(D d) public view returns (uint256) {
+        return d.x();
+    }
+}
+contract C {
+    D d;
+    constructor() public { d = new D(); }
+    function f() public view returns (uint256) {
+        return L.f(d);
+    }
+    function g() public returns (bool, uint256) {
+        (bool success, bytes memory data) = address(L).delegatecall(abi.encodeWithSelector(L.f.selector, d));
+        return (success, success ? abi.decode(data,(uint256)) : 0);
+    }
+    function h() public returns (bool, uint256) {
+        (bool success, bytes memory data) = address(L).call(abi.encodeWithSelector(L.f.selector, d));
```

Comment:
Would a test with `staticcall` also make sense here?

Reply:
Why? The only thing those are really meant to check is the delegatecall guard - that will apply to both call and staticcall just the same...

Reply:
Or will it? ;)

Reply:
Ok, fine with me